driver-core/src/main/com/mongodb/MongoClientSettings.java

```diff
@@ -771,12 +788,26 @@ public CodecRegistry getCodecRegistry() {
      *
      * @return the stream factory factory
      * @see Builder#streamFactoryFactory(StreamFactoryFactory)
+     * @see #getTransportSettings() ()
      */
+    @Deprecated
     @Nullable
     public StreamFactoryFactory getStreamFactoryFactory() {
         return streamFactoryFactory;
     }
 
+    /**
+     * Gets the settings for the underlying transport API
+     *
+     * @return the settings for the underlying transport API
+     *
+     * @since 4.11
+     */
+    @Nullable
+    public TransportSettings getTransportSettings() {
```

Comment:
Questions:

1. Is "transport" the right word for this?
2. Should this be part of SocketSettings instead?  (But then it's weird that NettySettings has an SslContext property that is more closely associated with SslSettings)
3. Is it weird that TransportSettings has no methods?